addons/account/tests/test_account_move_entry.py

```diff
@@ -134,6 +134,28 @@ def test_custom_currency_on_account_1(self):
 
         self.test_move.line_ids[0].account_id = custom_account
 
+    def test_fiscal_position(self):
+        """A move is assigned a fiscal position that matches its own company."""
+        company1 = self.company_data["company"]
+        company2 = self.company_data_2["company"]
+        partner = self.env['res.partner'].create({'name': 'Belouga'})
+        fpos1 = self.env["account.fiscal.position"].create(
+            {
+                "name": company1.name,
+                "company_id": company1.id,
+            }
+        )
+        fpos2 = self.env["account.fiscal.position"].create(
+            {
+                "name": company2.name,
+                "company_id": company2.id,
+            }
+        )
+        partner.with_company(company1).property_account_position_id = fpos1
+        partner.with_company(company2).property_account_position_id = fpos2
+        self.test_move.sudo().with_company(company2).partner_id = partner
+        self.assertEqual(self.test_move.fiscal_position_id, fpos1)
```

Comment:
I'm not sure I understand this part. You're creating the move with company2, but expected the invoice to have the fiscal position of company1 ?

Reply:
Thanks for the review!

To be precise, the move was created with the default company but I am assigning the partner as the OdooBot user while in a different company. This may seem contrived, but it emulates the native Odoo processes of creating invoices from incoming emails as well as bank statement entries in enterprise's account_online_synchronization when run as user OdooBot which is where we encountered the issue in our setup.

I expect the fiscal position to be assigned from the company of the move, not the company from the current environment.

Reply:
Understood, thanks for the clarification, I'll forward it to the final reviewer :)